### PR TITLE
chore(github): Update repo files

### PR DESCRIPTION
**Pull request by jmeridth**
- [x] DELETE `OWNERS` file, not parsed, only used for bookkeeping, `CODEOWNERS` does this
- [x] update `CODEOWNERS` to people who have committed in the last year
- [x] create `EMERITUS.md` to list former `CODEOWNERS` and thank them for their contribution
- [x] create `SECURITY.md` so people know how to report security issues
  - [x] turn on private vulnerability reporting in the repository
- [x] create `CODE_OF_CONDUCT.md` linking to CNCF code of conduct

By adding a code of conduct and security file we can set our `Community Standards` to 100% [here](https://github.com/argoproj/argo-helm/community)

Checklist:

* [x] I have bumped the chart version according to [versioning](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#versioning)
* [x] I have updated the documentation according to [documentation](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#documentation)
* [x] I have updated the chart changelog with all the changes that come with this pull request according to [changelog](https://github.com/argoproj/argo-helm/blob/main/CONTRIBUTING.md#changelog).
* [x] Any new values are backwards compatible and/or have sensible default.
* [x] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/blob/master/community/CONTRIBUTING.md).
* [x] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/stable/developer-guide/ci/)).

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

**Review by mbevc1 — APPROVED** [review]

File: CODEOWNERS [hunk 1]
```diff
@@ -1,16 +1,20 @@
...
+# All
+*                              @jmeridth
 
-# Other and new charts
-/charts/ @oliverbaehler
+# Github files
+.github/                       @mkilchhofer @jmeridth
+
+# All charts
+/charts/                       @jmeridth
```
Comment (mkilchhofer) [inline CODEOWNERS:8]: That will not work. Github does not merge the OWNERS with the more granular rules :(

Already tried that when I started contributing here: 
https://github.com/argoproj/argo-helm/pull/1238

File: OWNERS [hunk 2]
```diff
@@ -1,9 +1,3 @@
...
-- alexec
-- alexmt
-- jessesuen
-
-approvers:
-- alexec
-- alexmt
-- jessesuen
+- mkilchhofer
+- jmeridth
```
[the remark this answers is not in the document]
Comment (jmeridth) [reply OWNERS:3]: I'm good with removal.

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply OWNERS:3]: let's do it :)

[the remark this answers is not in the document]
Comment (mkilchhofer) [reply OWNERS:3]: pinged @crenshaw-dev on slack to verify no pipeline is using this file. Answer was:

> Right now, it’s just for bookkeeping
> afaik no actual function

File: CODEOWNERS [hunk 3]
```diff
@@ -1,16 +1,17 @@
...
+*                              @mkilchhofer @jmeridth
 
-# Other and new charts
-/charts/ @oliverbaehler
+# Github files
+.github/                       @mkilchhofer @jmeridth
 
 # Argo Workflows
-/charts/argo-workflows/ @stefansedich @paguos @vladlosev @yann-soubeyrand @jmeridth @yu-croco
+/charts/argo-workflows/        @vladlosev @yann-soubeyrand @yu-croco
```
Comment (mkilchhofer) [inline CODEOWNERS:8]:
```suggestion
/charts/argo-workflows/        @vladlosev @yann-soubeyrand @jmeridth @yu-croco
```

File: CODEOWNERS [hunk 4]
```diff
@@ -1,16 +1,17 @@
...
 
 # Argo CD
-/charts/argo-cd/ @davidkarlsen @mr-sour @yann-soubeyrand @mbevc1 @mkilchhofer @yu-croco @jmeridth @pdrastil
+/charts/argo-cd/               @mbevc1 @mkilchhofer @yu-croco @pdrastil
```
Comment (mkilchhofer) [inline CODEOWNERS:11]:
```suggestion
/charts/argo-cd/               @mbevc1 @mkilchhofer @yu-croco @jmeridth @pdrastil
```

File: CODEOWNERS [hunk 5]
```diff
@@ -1,16 +1,17 @@
-# https://help.github.com/en/github/creating-cloning-and-archiving-repositories/about-code-owners
+# All
+*                              @mkilchhofer @jmeridth
 
-# Other and new charts
-/charts/ @oliverbaehler
+# Github files
+.github/                       @mkilchhofer @jmeridth
```
Comment (mkilchhofer) [inline CODEOWNERS:5]: We can drop this as its already covered by the `*` rule :)

**Review by mkilchhofer — APPROVED** [review]
💯

**Pull request merged**